daemon/rm.go

```diff
@@ -36,19 +36,30 @@ func (daemon *Daemon) CleanPod(podId string) (int, string, error) {
 		}
 	}
 
-	os.RemoveAll(path.Join(utils.HYPER_ROOT, "services", podId))
-	pod.cleanupEtcHosts()
-	os.RemoveAll(path.Join(utils.HYPER_ROOT, "hosts", podId))
-
-	daemon.db.DeletePod(podId)
-	daemon.RemovePod(podId)
-	if pod.status.Type != "kubernetes" {
-		daemon.RemovePodContainer(pod)
+	pod.Lock()
+	defer pod.Unlock()
+	if pod.vm != nil {
+		pod.status.Status = types.S_POD_NONE
+		return code, cause, err
 	}
-	daemon.DeleteVolumeId(podId)
-	code = types.E_OK
 
-	return code, cause, nil
+	return daemon.RemovePodResource(pod)
+}
+
+func (daemon *Daemon) RemovePodResource(p *Pod) (int, string, error) {
+	os.RemoveAll(path.Join(utils.HYPER_ROOT, "services", p.id))
+	os.RemoveAll(path.Join(utils.HYPER_ROOT, "hosts", p.id))
+
+	daemon.db.DeletePod(p.id)
+	daemon.RemovePod(p.id)
+	if p.status.Type != "kubernetes" {
+		daemon.RemovePodContainer(p)
+	}
+	daemon.DeleteVolumeId(p.id)
+	code := types.E_OK
+
+	return code, "", nil
```

Comment:
ok, this is just copied from the original code.  

will update this, and are there any other comments.